gum/backend-qnx/gummemory-qnx.c

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright (C) 2008-2011 Ole André Vadla Ravnås <ole.andre.ravnas@tillitech.com>
+ *
+ * Licence: wxWindows Library Licence, Version 3.1
+ */
+
+#include "gummemory.h"
+#include "gumqnx-priv.h"
+
+#include <gio/gio.h>
+#include <fcntl.h>
+#include <sys/mman.h>
+#include <sys/procfs.h>
+#include <stdio.h>
+
+GumPageProtection
+_gum_page_protection_from_posix (const gint flags)
+{
+  GumPageProtection prot = GUM_PAGE_NO_ACCESS;
+
+  if (flags & PROT_READ)
+    prot |= GUM_PAGE_READ;
+  if (flags & PROT_WRITE)
+    prot |= GUM_PAGE_WRITE;
+  if (flags & PROT_EXEC)
+    prot |= GUM_PAGE_EXECUTE;
+
+  return prot;
+}
```

Comment:
PS: I noticed that the `backend-unix` implementation was inconsistent with these rules, so I fixed it when migrating it to `backend-posix`.